code/user_settings.py

```diff
@@ -0,0 +1,148 @@
+from talon import Module, fs, Context
+import os
+import csv
+from pathlib import Path
+from typing import Dict, List, Tuple
+import threading
+
+
+# NOTE: This method requires this module to be one folder below the top-level
+#   knausj folder.
+SETTINGS_DIR = Path(__file__).parents[1] / "settings"
+
+if not SETTINGS_DIR.is_dir():
+    os.mkdir(SETTINGS_DIR)
+
+
+mod = Module()
+ctx = Context()
+
+
+def _load_csv_dict(
+    file_name: str, headers=Tuple[str, str], default: Dict[str, str] = {}
+) -> Dict[str, str]:
+    """Load a word mapping from a CSV file. If it doesn't exist, create it."""
+    assert file_name.endswith(".csv")
+    path = SETTINGS_DIR / file_name
+
+    # Create the file if it doesn't exist
+    if not SETTINGS_DIR.is_dir():
+        os.mkdir(SETTINGS_DIR)
+    if not path.is_file():
+        with open(path, "w", encoding="utf-8") as file:
```

Comment:
Maybe dont write defaults to disk. Otherwise to remove a default it needs to be removed from the default list in the .py file and the csv file. This gets even more confusing with derived values which end up in multiple csv files.

Reply:
At the moment this is by design, but I can change that. I didn't want upstream defaults to affect the user's CSV files, so the defaults are only used once, when the CSV is initially created. If the user wants to delete them, `knausj_talon` won't override those changes. This creates a clean split between user CSVs and upstream knausj - you should never see a merge conflict and `settings` can be tracked independently.

Not sure what you mean by derived values - each CSV represents one Talon list in the global (thus top-level) context.

Reply:
Ah I see what you mean, and I read the code wrong the first time around. Thanks for the clarification!

Reply:
You dont need the fs watch at the end of the file if you instead use `talon.resource.open` here instead of open.